### PR TITLE
Fix Express route to allow any URL path and fix JSON body parsing

### PR DESCRIPTION
**Pull request by rcebulko**
Since endpoints can be `/r`, `/r-beta`, `/r-dev`, this PR just makes the main route a wildcard so it'll accept all requests.

Also adds @google-cloud/functions-framework for local testing

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -18,22 +18,36 @@ const express = require('express');
 const statusCodes = require('http-status-codes');
 
 const errorTracker = require('./routes/error-tracker');
-const querystring = require('./utils/requests/query-string');
```
Comment (jridgewell) [inline app.js:21]: I think we can delete this module now?

File: app.js [hunk 2]
```diff
@@ -18,22 +18,36 @@ const express = require('express');
...
 
 const app = express();
-const jsonParser = express.json({
-  limit: 10 * 1024 /* 10kb */,
-  type: () => true, // Attempt to allow any content-type.
-});
+const BODY_LIMIT = 10 * 1024 /* 10kb */;
+function rawJsonBodyParserMiddleware(req, res, next) {
+  if (!req.rawBody) {
+    // Defer to bodyParser when running as a server.
```
Comment (jridgewell) [inline app.js:27]: Do we even need this anymore?

File: app.js [hunk 3]
```diff
@@ -18,22 +18,36 @@ const express = require('express');
...
+    jsonParser = express.json({
+      limit: BODY_LIMIT,
+      type: () => true,
+    });
+    jsonParser(req, res, next);
+  } else if (req.rawBody.length > BODY_LIMIT) {
+    // When Cloud Functions hijacks the request, validate and parse it manually.
+    next(statusCodes.REQUEST_TOO_LONG);
+  } else {
+    req.body = JSON.parse(req.rawBody.toString());
```
Comment (jridgewell) [inline app.js:37]: Please do a length check to ensure we're under the body limit.

File: app.js (hunk 2 above)
Comment (rcebulko) [reply app.js:27]: Probably not tbh. We could probably get rid of the server module entirely, if we have no plans to redeploy to app engine

File: app.js (hunk 1 above)
Comment (rcebulko) [reply app.js:21]: It's still used to construct the query string for `querystring.stringify` we log to Stackdriver

File: app.js (hunk 3 above)
Comment (rcebulko) [reply app.js:37]: Do you mean something besides what's 4 lines up?

File: app.js (hunk 2 above)
Comment (rcebulko) [reply app.js:27]: Oh now I remember; the e2e tests are written using the express app as a standard Express server, and I haven't yet spent the time to rewrite the tests to use whatever test harness the Cloud Functions library (hopefully) provides

File: app.js (hunk 1 above)
Comment (jridgewell) [reply app.js:21]: Delete just `parse` then?

File: app.js (hunk 2 above)
Comment (jridgewell) [reply app.js:27]: I have no plans to app engine. Given that it's failed miserably during deploy, I'd consider it dead.

File: app.js (hunk 3 above)
Comment (jridgewell) [reply app.js:37]: Ohh, i missed that.

File: app.js (hunk 1 above)
Comment (rcebulko) [reply app.js:21]: Done.

File: app.js (hunk 2 above)
Comment (rcebulko) [reply app.js:27]: I think we had a race condition on my last comment. I'd like to punt on this for now and rewrite the e2e tests in a follow-up

Comment (jridgewell) [reply app.js:27]: That's fine. I don't imagine this will be difficult, we just need to install an extra `app.use()` helper in the test setup.

**Pull request merged**